api/src/main/java/org/openmrs/module/openconceptlab/client/OclClient.java

```diff
@@ -269,21 +336,70 @@ public long getTotalBytesToDownload() {
 	}
 	
 	public static class OclResponse {
-		
-		private final InputStream in;
-		
+
+
 		private final Date updatedTo;
-		
-		private final long contentLength;
-		
-		public OclResponse(InputStream in, long contentLength, Date updatedTo) {
-			this.in = in;
+
+		private long contentLength = 0;
+
+		private LinkedList<InputStream> inputStreams = new LinkedList<InputStream>();

```

Comment:
Use https://docs.oracle.com/javase/7/docs/api/java/util/Queue.html instead of LinkedList.